src/main/webapp/webapptests/spec/SpecGroups.js

```diff
@@ -0,0 +1,122 @@
+
+        it("should create result elements inside groupContainer", async function() {
+            spyOn(window, "fetch").and.returnValues(Promise.resolve({json: () => Promise.resolve(groups)}));
+
+            spyOn(document, "getElementById").and.returnValues(groupContainer);
+
+            await getSearchGroupResultsHelper(document, mockWindow);
+
+            expect(window.fetch).toHaveBeenCalledTimes(1);
+            expect(window.fetch.calls.allArgs()[0][0]).toEqual("/manage-groups?group=Test");
+
+            //one for the number of results label + 2 for the number of results in testResults
+            expect(groupContainer.childNodes.length).toEqual(3);
+            expect(groupContainer.childNodes[0].innerText).toContain("Found 2 groups named Test");
+        });
+
+    });
+
+    describe("when a group result is created", function() {
+        var result = {"name": "Test", "topic": "test", "description": "test", "id": "1"};
+        var element = createGroupResult(result);
+
+        it("should create div for result element", function() {
+            expect(element.tagName).toEqual("DIV");
+        });
+
+        it("should create h3 element inside result element for name", function() {
+            expect(element.childNodes[0].tagName).toEqual("H3");
+            expect(element.childNodes[0].innerText).toContain("Test");
+        });
+
+        it("should create h4 element inside result element for topic", function() {
+            expect(element.childNodes[1].tagName).toEqual("H4");
+            expect(element.childNodes[1].innerText).toContain("test");
+        });
+
+        it("should create h5 element inside result element for description", function() {
+            expect(element.childNodes[2].tagName).toEqual("H5");
+            expect(element.childNodes[2].innerText).toContain("test");
+        });
+
+        it("should create anchor element inside result element for group link", function() {
+            expect(element.childNodes[3].tagName).toEqual("A");
+            expect(element.childNodes[3].innerText).toContain("View");
+            expect(element.childNodes[3].href).toContain("/group.html?groupId=1");
+        });
+
+    });
+
+    describe("when user selects other in the topic selection form", function() {
+        var input = document.createElement("input");
+        input.id = "other-topic";
+
+        it("should display the input and make the input required", function() {
+            spyOn(document, "getElementById").and.returnValue(input);
+
+            checkOtherSelected("other");
+
+            expect(input.style.display).toBe("block");
+            expect(input.hasAttribute("required")).toBe(true);
+            expect(input.hasAttribute("aria-required")).toBe(true);
+            expect(input.getAttribute("aria-required")).toBe("true");
+        });
+    });
+
+    describe("when user had selected other topic but then selected something else", function() {
```

Comment:
If we put checkOtherSelected in general.js, we can delete the test here or in the MyLists tests.

Reply:
I'll delete the ones here and keep the ones in my-lists.js